### PR TITLE
fix: Remove useless notebook detection

### LINKED ISSUE #624: notebook // matplotlib confusion
The function in_notebook  is pretty useless, and it  prevents MatPlot to be in the qcodes name space, for no good reasons.
We do not really depend on the notebook anymore (thankfully).

### PR DESCRIPTION
**Pull request by giulioungaretti**
Fixes #624 .
@eendebakpt // @peendebak  this fixes our slack conversation too. 

@QCoDeS/core .

### CODE REVIEW

**Comment by jenshnielsen** [comment]
in_notebook is included in public.rst and should probably be removed from here too

```
Utils & misc
~~~~~~~~~~~~

.. autosummary::
   :toctree: generated/

   qcodes.utils.validators
   qcodes.utils.helpers.in_notebook
```

**Comment by giulioungaretti** [comment]
@jenshnielsen thanks for catching, zapped there too.

File: qcodes/__init__.py [hunk 1]
```diff
@@ -9,12 +9,10 @@
...
 from qcodes.version import __version__
-from qcodes.utils.helpers import in_notebook
 
-# code that should only be imported into the main (notebook) thread
-# in particular, importing matplotlib in the side processes takes a long
-# time and spins up other processes in order to try and get a front end
-if in_notebook():  # pragma: no cover
+
+if config.get("gui.plotlib"):
+    # load both because you never know what scientists actually want
```
Comment (MerlinSmiles) [inline qcodes/__init__.py]: I'd like an option for only one, if you want to be scientist proof and give it the default option 'both' with the possibility to only load one of them?

Comment (giulioungaretti) [inline qcodes/__init__.py]: cool!

Comment (giulioungaretti) [inline qcodes/__init__.py]: although, I bet, plenty will want both.  It feels pretty ugly to have a "Both", or "All" option.

**Comment by jenshnielsen** [comment]
replaced by #699

**Pull request closed without merging**